mbuild/compound.py

```diff
@@ -389,14 +403,14 @@ def rigid_particles(self, rigid_id=None):
             if particle.rigid_id is not None:
                 yield particle
 
-    def set_rigid(self, name=False, rigid_id=None):
+    def set_rigid(self, particle_name=False, rigid_id=None):
         """Treat the Compound and all successors as a single rigid body.
 
         Provides a unique rigid body ID to a Compound and its successors.
-        If 'name' is specified, only sucessors with this name are provided
-        with a rigid_id.  This is useful for Compounds where only some of
-        the particles should be rigid (such as a tethered nanoparticle with
-        a rigid core).
+        If 'particle_name' is specified, only sucessors with this name are
```

Comment:
That's a good point.  Actually the current code only sets the rigid_ids of those Compounds that share the provided `particle_name` (which I will change back to just `name`).  So if "CH2" was passed as the particle_name, only CH2 would get that rigid_id, but not its children.  This wouldn't be the desired behavior so I'll need to fix that, and will add another unit test.